packages/web-components/src/components/notice-choice/notice-choice.ts

```diff
@@ -0,0 +1,595 @@
+    loadContent(
+      cc,
+      lc,
+      (ncData) => {
+        this.ncData = ncData;
+        this.prepareCheckboxes();
+      },
+      (error) => {
+        console.error('error loading content', error);
+      }
+    );
+    if (this.country && this.country !== cc) {
+      this.countryChanged(this.country);
+    }
+  }
+  setDefaultSelections() {
+    if (!this.enableAllOptIn && this.checkboxes) {
+      const newValues = { ...this.values };
+      Object.keys(this.checkboxes).forEach((key) => {
+        const optInContent = this.optInContent || this.ncData.OptInContent;
+        const option = this._getOptionByQuestion(key, optInContent);
+        newValues[key] = !!(
+          option.checked === 'true' || option.checked === true
+        );
+        if (
+          this.defaultValues &&
+          Object.prototype.hasOwnProperty.call(this.defaultValues, key)
+        ) {
+          newValues[key] = this.defaultValues[key];
+        }
+        const fieldName = `NC_CHECK_${key}`;
+        const hiddenFieldName = `NC_HIDDEN_${key}`;
+        if (typeof this.onchange === 'function') {
+          this.onchange(fieldName, newValues[key]);
+          this.onchange(hiddenFieldName, newValues[key] ? 'OPT_IN' : null);
+        }
+      });
+      if (JSON.stringify(this.values) !== JSON.stringify(newValues)) {
+        this.values = newValues;
+      }
+    }
+  }
+  countryChangeAction() {
+    const splitValue = this.locale.split('-', 2);
+    const ncData = getNcContentFromWindow();
+    if (splitValue[1] === 'en') {
+      let preText = this.defaultPreText;
+      if (ncData.OtherPreferences.englishNoticeText !== '') {
+        preText = ncData.OtherPreferences.englishNoticeText;
+      }
+      this.preText = preText;
+    }
+    /**
+     * @description if the user already interacted with the checkboxes,
+     * skip country default selection.
+     */
+    if (!this.changed && !this.fetchedPref) {
+      /**
+       * @description
+       * change checkbox checked option based on new country.
+       */
+      this.setDefaultSelections();
+    }
+  }
+  countryChanged(newVal) {
+    const cc = newVal.toLocaleLowerCase();
+    const lc = countrySettings[newVal.toLocaleLowerCase()].lang;
+    resetToWorldWideContent();
+    loadContent(
+      cc,
+      lc,
+      (ncData) => {
+        /**
+         * @description Do not change content language.
+         * Change the checkbox according to the country rule.
+         */
+        this.optInContent = {
+          ...ncData.OptInContent,
+          cclc: `${worldWideContent.cc_name}-${worldWideContent.cc_lang}`,
+        };
+        this.countryChangeAction();
+      },
+      (error) => {
+        console.error('error loading content', error);
+      }
+    );
+  }
+  /**
+   *
+   * @param name name of the attribute
+   * @param oldVal old value of the attribute
+   * @param newVal new value of the attrbute
+   */
+  attributeChangedCallback(name, oldVal, newVal) {
```

Comment:
Is this and `_attributeChangedCallback` below being used? And if they are, it seems they're duplicated?